presto-docs/src/main/sphinx/develop/functions.rst

```diff
@@ -321,3 +321,66 @@ function follows:
   is used when performing a ``GROUP BY`` aggregation, and an implementation
   will be automatically generated for you, if you don't specify a
   ``AccumulatorStateFactory``
+
+
+Advanced Use Cases
+------------------
+
+Raw Block Inputs
+^^^^^^^^^^^^^^^^
+
+Both scalar and aggregation function annotations allow you to define methods
+which operate on native types. In Java, these native types are ``boolean``,
+``Slice``, and ``long``. However, for parameterized implementations or
+parametric types, the standard Java types can't always be used as they aren't
+able to fully represent the input data.
+
+It is possible to define a method handle which can accept *any* types by using
+``@BlockPosition`` in conjunction with the ``@BlockIndex`` parameters. This
+works for both scalar and aggregation function implementations. An example
+signature is provided below. 
+
+.. code-block:: java
+
+    @ScalarFunction("example")
+    public static Block exampleFunction(
+            @BlockPosition @SqlType("array(int)") Block block,
+            @BlockIndex int index) { /* ...implementation */ }
+
+Applying Generic Types with ``@BlockPosition``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Function signatures that use the ``@BlockPosition`` syntax are also able to
+operate over generic types when the function is defined with a
+``@TypeParameter`` annotation. Augment the ``@BlockPosition`` argument with an
```

Comment:
No suggestion, a compliment that this "Augment..." sentence is a great example of using imperative mood to clearly describe how and what the reader can do.

Reply:
Thanks steve!